scipy/sparse/construct.py

```diff
@@ -622,6 +622,85 @@ def bmat(blocks, format=None, dtype=None):
 
     return coo_matrix((data, (row, col)), shape=shape).asformat(format)
 
+def _block_diag(mats, format=None, dtype=None):
+    """      
+    Build a block diagonal sparse matrix from equally shaped matrices.
+    Much faster than block_diag for big matrices
+
+    Parameters
+    ----------
+    mats : sequence of equally shaped matrices
+        Input matrices.
+    format : str, optional
+        The sparse format of the result (e.g. "csr").  If not given, the matrix
+        is returned in "coo" format.
+    dtype : dtype specifier, optional
+        The data-type of the output matrix.  If not given, the dtype is
+        determined from that of `blocks`.
+
+    Returns
+    -------
+    res : sparse matrix
+
+    Notes
+    -----
+
+    .. versionadded:: 0.18.0
```

Comment:
Writing full docstrings for helper routines is not required, although it may be helpful.